src/WebApp/Extensions/Extensions.cs

```diff
@@ -95,4 +100,26 @@ public static void AddAuthenticationServices(this IHostApplicationBuilder builde
         var user = authState.User;
         return user.FindFirst("name")?.Value;
     }
+
+    public static void MapProductImageProxy(this IEndpointRouteBuilder app)
+    {
+        app.MapGet("/product-images/{id:int}", async (IHttpClientFactory httpClientFactory, int id) =>
+        {
+            var httpClient = httpClientFactory.CreateClient(ProductImageProxyHttpClient);
+            var imageResponse = await httpClient.GetAsync($"/api/v1/catalog/items/{id}/pic");
```

Comment:
Can/should Yarp be used for this? That's what "eshopLite" does: https://github.com/dotnet/aspire/blob/cb54d832ce8328c2ac681b031a8c0e0d0f709a7e/samples/eShopLite/MyFrontend/Program.cs#L31

Reply:
Yep

Reply:
Oh that's very nice. I started by trying to do this with Yarp but didn't discover the `MapForwarder` method, and got lost in the weeds with `LoadFromMemory`, trying to call `PathSet` and all kinds of mess. I did make it work in the end but then it was super unclear how to wire it up to service discovery, so I thought it was better not to introduce all that for this sample.

I'll have another go with `MapForwarder`. Is there an example of using this with service discovery somewhere? It needs to be able to make an outbound request to `http://catalogapi` so it needs to use an `HttpClient` that knows about this.

Reply:
> Is there an example of using this with service discovery somewhere? 

Yep! The same app uses the `Microsoft.Extensions.ServiceDiscovery.Yarp` package as well. See https://github.com/dotnet/aspire/blob/ab931f067195a7af0468c18974b864c51a2963cd/samples/eShopLite/MyFrontend/Program.cs#L9

Reply:
Excellent, thanks. It works perfectly - this PR now uses that.